### PR TITLE
Improving examples | Cogs (Not extensions)

### LINKED ISSUE #2525: Improving examples
Right now, we don't have a lot of examples that show how to work with various parts of the library. I'm hoping in the near future to revamp examples to be a bit more useful. Part of the on-going work in doing so is to document the current examples (or maybe even rewrite them) significantly through commenting most parts of the code as if someone is explaining it to you over your shoulder.

Hopefully by commenting the code extensively the examples are more helpful for those who learn primarily through examples.

Likewise, I want to add examples that showcase certain things:

- [ ] Subcommands
- [ ] Various types of converters
- [ ] Basic reaction roles
- [ ] HelpCommand overriding
- [ ] Cogs (not extensions)
- [ ] Extensions (using cogs)
- [ ] Custom contexts
- [ ] Working with permissions (reading and writing, e.g. roles)
- [ ] Working with permission overwrites (channel overwrites etc)
- [ ] How to make a secret channel
- [ ] Error handling via ext.commands
- [ ] How to use AutoShardedBot/AutoShardedClient
- [ ] An example showcasing how to use the task extension

More example suggestions are welcome. I also would want the previous examples to be heavily commented as well. We can draw inspiration from [serenity's examples](https://github.com/serenity-rs/serenity/tree/current/examples/).

One thing that remains is whether these examples should be linear, similar to a book with chapters where we build upon knowledge on the previous examples. That way we don't have to overly comment everything if it's been covered in a previous example.

### PR DESCRIPTION
**Pull request by bleudev**
## Summary

Add example with cogs (not extensions) from this issue #2525
(fix #2525)

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by GoogleGenius** [comment]
Please run this through black.

File: examples/cogs/basic_not_extension.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+            await message.reply('Pong!')
+
+    @commands.bot_has_permissions(administrator=True)
+    @commands.command()
+    async def admin_check(self, ctx: commands.Context):
+        await ctx.send('You are admin!')
+
+# Create bot with cogs
+class Bot(commands.Bot):
+    def __init__(
```
Comment (GoogleGenius) [inline examples/cogs/basic_not_extension.py:29]: Unnecessary...

Comment (ArthurKun021) [reply examples/cogs/basic_not_extension.py:29]: I personally like having `_init_` there so you I can use the type hinting for the when I initialized some variables

**Comment by pixeldeee** [comment]
> Please run this through black.

Ok

File: examples/cogs/basic_not_extension.py (hunk 1 above)
Comment (pixeldeee) [reply examples/cogs/basic_not_extension.py:29]: > Unnecessary to override the base class `__init__`

Maybe remove this method?

File: examples/cogs/basic_not_extension.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            help_command=help_command,
+            tree_cls=tree_cls,
+            description=description,
+            intents=intents,
+            **options
+        )
+
+    async def on_ready(self):  # Install cogs
+        print('Start installing cogs!')
+        await self.add_cog(MyCog(self))
```
Comment (LucasCoderT) [inline examples/cogs/basic_not_extension.py:51]: It's heavily recommended to use `setup_hook` to do this.

Comment (pixeldeee) [reply examples/cogs/basic_not_extension.py:51]:
```suggestion
    async def setup_hook(self):
        await self.add_cog(MyCog(self))
```
This code?

Comment (GoogleGenius) [reply examples/cogs/basic_not_extension.py:51]: yes

**Comment by pixeldeee** [comment]
@Rapptz pls review this pull

**Pull request closed without merging**